src/Shared/EncodingUtilities.cs

```diff
@@ -148,5 +155,92 @@ internal static bool FileStartsWithPreamble(string file)
                 return StartsWithPreamble(stream);
             }
         }
+
+        /// <summary>
+        /// Checks to see if a string can be encoded in a specified code page.
+        /// </summary>
+        /// <remarks>Internal for testing purposes.</remarks>
+        /// <param name="codePage">Code page for encoding.</param>
+        /// <param name="stringToEncode">String to encode.</param>
+        /// <returns>True if the string can be encoded in the specified code page.</returns>
+        internal static bool CanEncodeString(int codePage, string stringToEncode)
+        {
+            // We have a System.String that contains some characters. Get a lossless representation
+            // in byte-array form.
+            var unicodeEncoding = new UnicodeEncoding();
+            var unicodeBytes = unicodeEncoding.GetBytes(stringToEncode);
+
+            // Create an Encoding using the desired code page, but throws if there's a
+            // character that can't be represented.
+            var systemEncoding = Encoding.GetEncoding(codePage, EncoderFallback.ExceptionFallback,
+                DecoderFallback.ExceptionFallback);
+
+            try
+            {
+                var oemBytes = Encoding.Convert(unicodeEncoding, systemEncoding, unicodeBytes);
```

Comment:
Curious, is this different from/better than calling `systemEncoding.GetBytes(stringToEncode)`?

Oh, this was moved from elsewhere.

Reply:
Looks like it's pretty much just that: https://github.com/dotnet/runtime/blob/843d66b238817c7bdbe42556bb17ab1e9f10973a/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs#L202